ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/requests/RequestProcessorCoordinator.java

```diff
@@ -41,12 +41,10 @@ public Optional<List<Request>> process(final ProcessRequestContext context) {
     List<Request> requests = null;
     for (RequestProcessor requestProcessor : processors.values()) {
       var r = requestProcessor.process(context);
-      if (r.isPresent()) {
-        if (requests == null) {
-          requests = new ArrayList<>();
-        }
-        requests.addAll(r.get());
+      if (requests == null) {
+        requests = new ArrayList<>();
       }
+      requests.add(r);
```

Comment:
With this change, the only way to return `Optional.Empty ` is a processor coordinator with no processors. We can throw an exception in the builder if trying to build with empty processors and simplify this method to return List<Requests>

Reply:
nice suggestion. I've done that and also simplified the process method to use a stream because doesn't need to consider nulls now